### ISSUE TITLE
route blinding: use all valid routes during path construction iteration instead of the currently capped set

### ISSUE THREAD

**Issue by ellemouton** [lightningnetwork/lnd#9076]
Today, if we are constructing set of blinded payment paths to add to an invoice, we: 

1) [Select up to `MaxNumPaths` routes](https://github.com/lightningnetwork/lnd/blob/a895b1c6caf00518ec41fe44a84288160f52a264/routing/router.go#L703-L712) to our own node that are considered blinded path candidates. 
2) This set is then passed to `blindedpath.BuildBlindedPaymentPaths` which attempts to use each path to build a valid blinded path
3) If `blindedpath.BuildBlindedPaymentPaths` finds that it cannot use one of the paths it was given, it [skips it and moves on to the next one](https://github.com/lightningnetwork/lnd/blob/a895b1c6caf00518ec41fe44a84288160f52a264/routing/blindedpath/blinded_path.go#L152-L155)

This order of operations means that we may run out of paths in `blindedpath.BuildBlindedPaymentPaths` and we wont go back and try some of the paths that we initially found by [`ChannelRouter`'s `findBlindedPaths` ](https://github.com/lightningnetwork/lnd/blob/a895b1c6caf00518ec41fe44a84288160f52a264/routing/router.go#L625). 

What we instead want is for `blindedpath.BuildBlindedPaymentPaths` to be able to continue with the iteration through possibly _all_ of the routes found in `findBlindedPaths` until it reaches the max specified by `MaxNumPaths`.

**Comment by ziggie1984** [comment]
Good idea, so you mean we should immediately when having the blinded path check if we can build the whole route before exiting the route search, so we make sure the returned routes are also possible to use ?

**Comment by ellemouton** [comment]
@ziggie1984  -  yeah, or we make a way for the logic that is doing the iterating and building to query like `NextCandidateRoute` or something like that. Or actually, perhaps should return all possible routes and then do the MaxPath cap later on instead of as early as it is currently done

**Comment by ellemouton** [comment]
@MPins  - im saying it currently does the following: 

1) Chooses MaxNumPaths candidate paths
2) Passes that to BuildBlindedPaths which might drop some of the paths returned cause they turn out not to be valid. 

This is not ideal cause we then we can easily run out of paths to try. 

Instead, the MaxNumPaths restriction should be applied later on instead. So when selecting candidates, we should not limit at that point.

**Comment by MPins** [comment]
Got it. If nobody had started working on it, I can give a try.

**Comment by saubyk** [comment]
This is for my education...why do we have a limit capped by `MaxNumPaths`?

**Comment by ziggie1984** [comment]
otherwise the invoice would become super huge, there might be a huge number of paths to a particular node.

**Comment by litbot-9000** [comment]
Closing as duplicate of #8363 which tracks support for multiple blinded paths in invoices.
